x-pack/legacy/plugins/siem/server/lib/detection_engine/alerts/__mocks__/es_results.ts

```diff
@@ -124,9 +209,9 @@ export const repeatedSearchResultsWithSortId = (repeat: number) => ({
     skipped: 0,
   },
   hits: {
-    total: repeat,
+    total,
     max_score: 100,
-    hits: Array.from({ length: repeat }).map(x => ({
+    hits: Array.from({ length: pageSize }).map(x => ({
```

Comment:
I see here that this is where you want the auto generated guid's probably. What you can do is create a small array of pre-generated GUID's with your tests and then reference them here through your map by cloning the sampleDocWithSortId and then using the the index argument to map to change the `_id` as needed.

I see that you never go above 4 which is great, as we shouldn't be testing too large of data sets in unit tests or writing too much test harness code. That's typically better for the larger e2e tests to catch.

A pre-generated array of 4 unique id's and then changing them out here would be ideal. Then you get the same predictable 4 unique id's and can reference them if you're doing expects.

Reply:
Nice! That's a good idea. I like how the addition of guids generated within the test instead of in the sample data reduces the "moving parts" component that can result in flaky tests too.